amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectionSettings.scala

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.amqp
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.rabbitmq.client.{Address, Connection, ConnectionFactory, ExceptionHandler}
+
+/**
+ * Only for internal implementations
+ */
+sealed trait AmqpConnectionSettings {
+  def getConnection: Connection
+}
+
+final case class ReusableAmqpConnectionSettings(settings: AmqpConnectionSettings) extends AmqpConnectionSettings {
```

Comment:
I'd make this take a `Connection`, not have any vars and call it `ExternalConnectionProvider` or something along those lines. And then have it do nothing on `release` (because we just moved that responsibility to the user)

Reply:
The whole idea of taking the setting is that you can release it and then reuse in a different stream and it will create another connection.

If you pass a connection and that connection is closed then the instance becomes unusable and unrecoverable.

Don't you think that's a valid case?